.github/workflows/image-build-ci.yml

```diff
@@ -28,6 +34,22 @@ jobs:
         with:
           arguments: pluginZip
 
+      - name: Set version variable - on push = branch name; on PR = target branch name
+        shell: bash
+        run: |
+          # Short name for current branch. For PRs, use target branch (base ref)
+          GIT_BRANCH=${GITHUB_BASE_REF:-${GITHUB_REF#refs/heads/}}
+          echo "GIT_BRANCH=$GIT_BRANCH" >> $GITHUB_ENV
+
+      - name: Prepare Dockerfile
+        run: |
+          cp Dockerfile.template Dockerfile
+          sed -i 's/{version}/${{env.GIT_BRANCH}}/' Dockerfile
```

Comment:
Could be jest one step `sed 's/{version}/${{env.GIT_BRANCH}}/' Dockerfile.template > Dockerfile`